WooCommerce/Classes/Tools/CardPresentPluginsDataProvider.swift

```diff
@@ -30,16 +30,18 @@ protocol CardPresentPluginsDataProviderProtocol {
 struct CardPresentPluginsDataProvider: CardPresentPluginsDataProviderProtocol {
     private let storageManager: StorageManagerType
     private let stores: StoresManager
-    private let configuration: CardPresentPaymentsConfiguration
+    private let configurationLoader: CardPresentConfigurationLoader
+
+    static let shared = CardPresentPluginsDataProvider(configurationLoader: CardPresentConfigurationLoader(stores: ServiceLocator.stores))
```

Comment:
Hey, good question. `CardPresentPluginsDataProvider` only holds `CardPresentConfigurationLoader` which in turn doesn't keep any state and returns a configuration. However, since it's not required we could do without this singleton.

Reply:
If we can do without it, I think we should. I don't think it causes any issues now, but it's the kind of thing which might in future as the code is changed. 

The StripeCardReaderService is already a singleton, so having more than one interacting with each other could get quite messy. Would you mind changing it?

Reply:
Of course. I changes it here:

https://github.com/woocommerce/woocommerce-ios/pull/13982

I also retested `PaymentReceiptEmailParameterDeterminer` on `trunk` to make sure there are no issues even without a fix.